widget/editor.go

```diff
@@ -99,6 +112,39 @@ type maskReader struct {
 	overflow []byte
 }
 
+// combinedPos is a point in the editor, three different ways.
+type combinedPos struct {
+	// editorBuffer offset, from e.rr.caret. The other three fields are based
+	// off of this one.
+	ofs int
+
+	// lineCol.Y = line (offset into e.lines), and X = col (offset into
+	// e.lines[Y])
+	lineCol screenPos
+
+	// Pixel coordinates
+	x fixed.Int26_6
+	y int
+}
+
+type SelectionAction int
+
+const (
+	SelectionDoNotChange SelectionAction = iota
```

Comment:
I understand SelectionDoNotChange may be needed internally, but when would you use it externally? It seems to me that the selection should always be updated somehow when moving the caret.

Perhaps the caret should be defined as a "fat caret" (start, end) and Editor should have two move methods, MoveCaret and MoveCaretEnd?